dockstore-integration-testing/src/test/java/io/dockstore/client/cli/GeneralIT.java

```diff
@@ -727,6 +715,67 @@ public void testGrabbingImagesFromQuay() {
         assertEquals(count, count2);
     }
 
+    @Test
+    public void testGrabChecksumFromDockerHub() {
```

Comment:
For quay.io and DockerHub I'd like to see a test where
1) A tool has tags with existing records for checksums
2) We do a refresh, but quay.io/DockerHub are slow and/or non-responsive
3) Verify that we just keep the existing checksums

Reply:
So for dockerhub and quay, it's not getting erased and now added hacky tests for that.
For quay, I am using the same call we already had to grab the tags to get the image information. If the response from quay is empty/times out, then all the tags will get deleted on Dockstore. Believe this was the behaviour before I made my changes.

Discussed with Charles and if the response from Quay is empty, then it makes sense to delete the tags, but not if Quay is being unresponsive/is down. Can make new ticket if you agree.

Reply:
I agree, that sounds reasonable.